doc/nrf/security/ap_protect.rst

```diff
@@ -173,8 +173,16 @@ Based on the available implementation types, you can configure the access port p
 Enabling AP-Protect with :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK`
 ====================================================================
 
-Setting the :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware is enough to enable the access port protection mechanism for SoCs of the nRF53 Series and those SoCs of the nRF52 Series that feature the hardware and software type of AP-Protect.
-The access port protection configured in this way cannot be disabled without erasing the flash.
+Setting the :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware enables the access protection mechanism for SoCs of the nRF53 Series and those SoCs of the nRF52 Series that feature the hardware and software type of AP-Protect.
+
+The setting writes the debugger register in SystemInit() in order to lock the access port protection in every boot. In addition to this, the UICR.APPROTECT register should also be written as instructed in `app_approtect_uicr_approtect`_.
+
+.. note::
+    With multi-image builds, this setting needs to be set for the first image (usually a bootloader). Otherwise the AP-Protect will not be sufficient and debugger can be attached to the device.
+    With sysbuild, use the SB_CONFIG_APPROTECT_LOCK, which sets this for all the images.
+
+.. important::
+    With nRF91x1 series devices, the registry setting related to :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` does not persist in System ON IDLE mode. You MUST lock the UICR.APPROTECT register to enable the AP-Protect mechanism as instructed in `app_approtect_uicr_approtect`_.
```

Comment:
@SeppoTakalo @anangl: Instead of this, I would rather have the hal to lock the UICR.APPROTECT for nRF91x1 devices, if this Kconfig is set. And we could document that instead. Same goes for CONFIG_NRF_SECURE_APPROTECT_LOCK.

Reply:
After thinking about this, maybe we should just document it.
As we now fix the documentation, then it means that the documentation is correct for most of the NCS releases. But if you instead fix the code, it might be that customer reading the documentation of new release might be miss-guided when they would use older SDK that does not lock automatically.

So I think it is best to leave the documentation like you suggest.

Reply:
I think that if we changed the code, we could add a comment:
"From NCS 2.8 onwards, the :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` locks the UICR.APPROTECT register in SystemInit and resets the device.".

But I am fine leaving this as documented at this point.